source/backend/api/Services/FinancialCodeService.cs

```diff
@@ -133,7 +152,93 @@ public FinancialCodeModel Add(FinancialCodeTypes type, FinancialCodeModel model)
                     }
 
                 default:
-                    throw new BadRequestException("Financial code type not valid.");
+                    throw new BadRequestException("Financial code type must be a known type");
+            }
+        }
+
+        public FinancialCodeModel Update(FinancialCodeTypes type, FinancialCodeModel model)
+        {
+            model.ThrowIfNull(nameof(model));
+
+            _logger.LogInformation("Updating financial code with type {type} and model {model}", type, model);
+            User.ThrowIfNotAuthorized(Permissions.SystemAdmin);
+
+            ValidateVersion(type, model.Id, model.RowVersion);
```

Comment:
What is the benefit of checking this manually? The db will throw a dbconcurrency exception if the rowversion is not updated.

Reply:
Well, this pattern is pervasive across other services. If we want to abandon it then we should do it across all of them. Topic for Friday chat I think

Reply:
This is a snap of another service doing the check.
![image](https://user-images.githubusercontent.com/24322224/215862851-53586a0f-4e1d-42a0-805b-29412a1f1d7f.png)

Reply:
I think the original intent of that method was to be used by child entities to make sure the parent was in sync